File: PBOLanjut_Pertemuan13/Tugas/client/Warnet.py
import requests
import json

class sewa:

    def __init__(self):
        self.__id=None
        self.__idpc=None
        self.__user=None
        self.__tanggal=None
        self.__jam_mulai=None
        self.__jam_selesai=None
        self.__lama_waktu=None
        self.__tarif=None
        self.__total=None
        self.__status=None
        self.__url="http://localhost/webapi/warnet_api.php"
        
        
    @property
    def id(self):
        return self.__id

    @property
    def idpc(self):
        return self.__idpc

    @idpc.setter
    def idpc(self, value):
        self.__idpc = value

    @property
    def user(self):
        return self.__user

    @user.setter
    def user(self, value):
        self.__user = value

    @property
    def tanggal(self):
        return self.__tanggal

    @tanggal.setter
    def tanggal(self, value):
        self.__tanggal = value

    @property
    def jam_mulai(self):
        return self.__jam_mulai

    @jam_mulai.setter
    def jam_mulai(self, value):
        self.__jam_mulai = value
        
    @property
    def jam_selesai(self):
        return self.__jam_selesai

    @jam_selesai.setter
    def jam_selesai(self, value):
        self.__jam_selesai = value
        
    @property
    def lama_waktu(self):
        return self.__lama_waktu

    @lama_waktu.setter
    def lama_waktu(self, value):
        self.__lama_waktu = value
        
    @property
    def tarif(self):
        return self.__tarif

    @tarif.setter
    def tarif(self, value):
        self.__tarif = value
        
    @property
    def total(self):
        return self.__total

    @total.setter
    def total(self, value):
        self.__total = value
        
    @property
    def status(self):
        return self.__status

    @status.setter
    def status(self, value):
        self.__status = value

    def getByidpc(self, idpc):
        url = self.__url+"?idpc="+idpc
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id=item["id"]
            self.__idpc=item["idpc"]
            self.__user=item["user"]
            self.__tanggal=item["tanggal"]
            self.__jam_mulai=item["jam_mulai"]
            self.__jam_selesai=item["jam_selesai"]
            self.__lama_waktu=item["lama_waktu"]
            self.__tarif=item["tarif"]
            self.__total=item["total"]
            self.__status=item["status"]
        return data

    def simpan(self):
        payload = {
            "idpc":self.__idpc,
            "user":self.__user,
            "tanggal":self.__tanggal,
            "jam_mulai":self.__jam_mulai,
            "jam_selesai":self.__jam_selesai,
            "lama_waktu":self.__lama_waktu,
            "tarif":self.__tarif,
            "total":self.__total,
            "status":self.__status
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    
    def updateByidpc(self, idpc):
        url = self.__url+"?idpc="+idpc
        payload = {
            "idpc":self.__idpc,
            "user":self.__user,
            "tanggal":self.__tanggal,
            "jam_mulai":self.__jam_mulai,
            "jam_selesai":self.__jam_selesai,
            "lama_waktu":self.__lama_waktu,
            "tarif":self.__tarif,
            "total":self.__total,
            "status":self.__status
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    
    def getAllData(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    
    def deleteByidpc(self,idpc):
        url = self.__url+"?idpc="+idpc
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: PBOLanjut_Pertemuan13/Tugas/webapi/Warnet.php
<?php
require_once 'database.php';

class Warnet
{
    private $db;
    private $table ='sewa';
    public $idpc = "";
    public $user = "";
    public $tanggal = "";
    public $jam_mulai = "";
    public $jam_selesai = "";
    public $lama_waktu = "";
    public $tarif = "";
    public $total = "";
    public $status = "";

    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }

    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }

    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }

    public function get_by_idpc(int $idpc)
    {
        $query = "SELECT * FROM $this->table WHERE idpc = $idpc";
        $result_set = $this->db->query($query);   
        return $result_set;
    }

    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`idpc`, `user`, `tanggal`,`jam_mulai`,`jam_selesai`,`lama_waktu`,`tarif`,`total`,`status`) VALUES ('$this->idpc','$this->user','$this->tanggal','$this->jam_mulai','$this->jam_selesai','$this->lama_waktu','$this->tarif','$this->total','$this->status')";
        $this->db->query($query);
        return $this->db->insert_id();
    }

    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET `idpc` = '$this->idpc', `user` = '$this->user', `tanggal` = '$this->tanggal', `jam_mulai` = '$this->jam_mulai', `jam_selesai` = '$this->jam_selesai', `lama_waktu` = '$this->lama_waktu', `tarif` = '$this->tarif', `total` = '$this->total', `status` = '$this->status' WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }

    public function update_by_idpc($idpc): int
    {
        $query = "UPDATE $this->table SET `user` = '$this->user', `tanggal` = '$this->tanggal', `jam_mulai` = '$this->jam_mulai', `jam_selesai` = '$this->jam_selesai', `lama_waktu` = '$this->lama_waktu', `tarif` = '$this->tarif', `total` = '$this->total', `status` = '$this->status' WHERE idpc = $idpc";
        $this->db->query($query);
        return $this->db->affected_rows();
    }

    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }

    public function delete_by_idpc($idpc): int
    {
        $query = "DELETE FROM $this->table WHERE idpc = $idpc";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>